crates/burn-fusion/src/client.rs

```diff
@@ -281,3 +297,33 @@ where
         server.resolve_server_bool::<B>(&tensor.into_ir())
     }
 }
+
+/// Extension trait to extract outputs when registering an operation.
+pub trait OperationOutput<R: FusionRuntime> {
+    /// Extract a single output tensor.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the operation produced more or fewer than one output.
+    fn output(self) -> FusionTensor<R>;
+
+    /// Extract a pair of output tensors.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the operation produced more or fewer than two outputs.
+    fn outputs(self) -> (FusionTensor<R>, FusionTensor<R>);
+}
```

Comment:
I would change that pattern to the following:

```rust
use std::path::PathBuf;

trait Output {
    fn outputs<const N: usize>(self) -> [String; N];
}

impl Output for Vec<PathBuf> {
    fn outputs<const N: usize>(self) -> [String; N] {
        let arra: [PathBuf; N] = self.try_into().unwrap();
        return arra.map(|path| format!("{:?}", path));
    }
}
```

That is a simple example, but it generalize to any length and it determined at compile time.